output.tf

```
@@ -14,6 +14,6 @@ output "port" {
 }
 
 output "host" {
-  value       = "${module.dns.hostname}"
+  value       = "${element(coalescelist(compact(list(module.dns.hostname)), aws_elasticache_replication_group.default.*.primary_endpoint_address), 0)}"
```

```suggestion
  value       = "${coalesce(module.dns.hostname, join("", aws_elasticache_replication_group.default.*.primary_endpoint_address)}"
```
If `enabled=false`, `module.dns.hostname` will be empty and `aws_elasticache_replication_group.default.*.primary_endpoint_address` will be empty list.
does `element` work on an empty list?

We can simplify by using `coalesce` and `join`.

---

Thanks, that's a good point,